### ISSUE TITLE
[build] boost::none changes (1.60.0) break build

### ISSUE THREAD

**Issue by gicmo** [G-Node/nix-java#62]
See issue g-node/nix#570 for details.

Error message for the laughs: 
```bash
jniNone.cpp:1087:33: error: no matching constructor for initialization of 'nix::none_t' (aka 'boost::none_t')
        nix::none_t* rptr = new nix::none_t();
                                ^
/usr/local/include/boost/none_t.hpp:32:12: note: candidate constructor not viable: requires 1 argument, but 0 were provided
  explicit none_t(init_tag){} // to prevent default constructor
           ^
/usr/local/include/boost/none_t.hpp:29:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 0 were provided
struct none_t
       ^
/usr/local/include/boost/none_t.hpp:29:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 0 were provided
```

**Comment by gicmo** [comment]
I applied a bad-aid to see if I can fix the rest to get it compile and work on OS X (see #61), but I don't think that this is the canonical way to fix it:

```diff
diff --git i/src/main/java/org/g_node/nix/internal/None.java w/src/main/java/org/g_node/nix/internal/None.java
index 62ef194..ffcc445 100644
--- i/src/main/java/org/g_node/nix/internal/None.java
+++ w/src/main/java/org/g_node/nix/internal/None.java
@@ -2,24 +2,29 @@ package org.g_node.nix.internal;

 import org.bytedeco.javacpp.Loader;
 import org.bytedeco.javacpp.Pointer;
-import org.bytedeco.javacpp.annotation.Name;
-import org.bytedeco.javacpp.annotation.Namespace;
-import org.bytedeco.javacpp.annotation.Platform;
-import org.bytedeco.javacpp.annotation.Properties;
+import org.bytedeco.javacpp.annotation.*;

 /**
  * <h1>None</h1>
  * Low level wrapper to NIX None.hpp .
  */

-@Properties(value = {
-        @Platform(include = {"<nix/None.hpp>"}),
-        @Platform(value = "linux"),
-        @Platform(value = "windows")})
+
+
+@Platform(include = {"<nix/None.hpp>"})
 @Namespace("nix")
 @Name("none_t")
 public class None extends Pointer {

+    @Name("init_tag")
+    class InitTag extends Pointer {
+        public InitTag() {
+            allocate();
+        }
+
+        private native void allocate();
+    }
+
     static {
         Loader.load();
     }
@@ -28,8 +33,8 @@ public class None extends Pointer {
      * Constructor
      */
     public None() {
-        allocate();
+        allocate(new InitTag());
     }

-    private native void allocate();
+    private native void allocate(@ByVal InitTag tag);
 }
```

**Comment by gicmo** [comment]
Should be fixed with #69
